src/Component/src/Metadata/Resource/Factory/PluralNameResourceMetadataCollectionFactory.php

```diff
@@ -0,0 +1,82 @@
+<?php
+
+/*
+ * This file is part of the Sylius package.
+ *
+ * (c) Sylius Sp. z o.o.
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+declare(strict_types=1);
+
+namespace Sylius\Resource\Metadata\Resource\Factory;
+
+use Sylius\Resource\Exception\LogicException;
+use Sylius\Resource\Metadata\Inflector\Inflector;
+use Sylius\Resource\Metadata\Inflector\InflectorInterface;
+use Sylius\Resource\Metadata\MetadataInterface;
+use Sylius\Resource\Metadata\RegistryInterface;
+use Sylius\Resource\Metadata\Resource\ResourceMetadataCollection;
+use Sylius\Resource\Metadata\ResourceMetadata;
+
+/**
+ * @experimental
+ */
+final class PluralNameResourceMetadataCollectionFactory implements ResourceMetadataCollectionFactoryInterface
+{
+    private InflectorInterface $inflector;
+
+    public function __construct(
+        private readonly ResourceMetadataCollectionFactoryInterface $decorated,
+        ?InflectorInterface $inflector = null,
+        private readonly bool $routingBcLayerEnabled = true,
+        private readonly ?RegistryInterface $resourceRegistry = null,
```

Comment:
The first 2 don't seem to need the default values; the last one is questionable, though it is set within the service definition.
Is it possible to not have the resource registry declared?